cms/plugin_rendering.py

```diff
@@ -125,12 +125,30 @@ def render_placeholder(placeholder, context_to_copy, name_fallback="Placeholder"
     request = context['request']
     if not hasattr(request, 'placeholders'):
         request.placeholders = {}
-    perms = (placeholder.has_change_permission(request) or not placeholder.cache_placeholder)
-    if not perms or placeholder.slot not in request.placeholders:
-        request.placeholders[placeholder.slot] = (placeholder, perms)
+
+    # Prepend frontedit toolbar output if applicable
```

Comment:
Yeah, just avoived cluttering the PR  with other optimization at first try
Code reworked now